test/rich-text/plugins/link-paste-handler.test.ts

```diff
@@ -0,0 +1,145 @@
+            expect(insertedNode.text).toBe(text);
+            expect(insertedNode.marks).toHaveLength(0); // not a link
+        }
+    );
+
+    it.each(URLTestData)(
+        "should handle pasting URL text without existing selection (%#)",
+        (text) => {
+            const view = createView(createState("", [linkPasteHandler]));
+
+            dispatchPasteEvent(view.dom, {
+                "text/plain": text,
+            });
+
+            const insertedNode = view.state.doc.nodeAt(
+                view.state.selection.from - 1
+            );
+
+            expect(insertedNode.isText).toBe(true);
+            expect(insertedNode.text).toBe(text);
+            expect(insertedNode.marks).toHaveLength(1);
+            expect(insertedNode.marks[0].type.name).toBe("link");
+            expect(insertedNode.marks[0].attrs.href).toBe(text);
+            expect(insertedNode.marks[0].attrs.markup).toBe("linkify");
+            expect(insertedNode.marks[0].attrs.title).toBeNull();
+        }
+    );
+
+    it.each(URLTestData)(
+        "should use existing selection as link text (%#)",
+        (text) => {
+            let state = createState("<p>my example link</p>", [
+                linkPasteHandler,
+            ]);
+            state = applySelection(state, 0, 15);
+            const view = createView(state);
+
+            dispatchPasteEvent(view.dom, {
+                "text/plain": text,
+            });
+
+            const insertedNode = view.state.doc.nodeAt(
+                view.state.selection.from - 1
+            );
+
+            expect(insertedNode.isText).toBe(true);
+            expect(insertedNode.text).toBe("my example link");
+            expect(insertedNode.marks).toHaveLength(1);
+            expect(insertedNode.marks[0].type.name).toBe("link");
+            expect(insertedNode.marks[0].attrs.href).toBe(text);
+            expect(insertedNode.marks[0].attrs.markup).toBeNull();
+            expect(insertedNode.marks[0].attrs.title).toBeNull();
+        }
+    );
+
+    it.each(URLTestData)(
+        "should gracefully paste into inline code (%#)",
+        (text) => {
+            let state = createState("<code>int i = 5;</code>", [
+                linkPasteHandler,
+            ]);
+            state = applySelection(state, 3);
+            const view = createView(state);
+
+            dispatchPasteEvent(view.dom, {
+                "text/plain": text,
+            });
+
+            const node = view.state.doc.nodeAt(view.state.selection.from - 1);
+
+            expect(node.isText).toBe(true);
+
+            expect(node.text).toBe(`int${text} i = 5;`);
+            expect(node.marks).toHaveLength(1);
+            expect(node.marks[0].type.name).toBe("code");
+        }
+    );
+
+    it.each(URLTestData)(
+        "should gracefully paste into a code block (%#)",
+        (text) => {
+            let state = createState(
+                `<pre>int i = 5;
+i++;
+Console.WriteLine(i);</pre>`,
```

Comment:
nit: I think this might need `<code>` for the schema to detect as a `code_block`.

```suggestion
                `<pre><code>int i = 5;
i++;
Console.WriteLine(i);</code></pre>`,
```

optional: When I'm testing behavior against very specific nodes, I tend to add a quick `expect` to make sure the nodes I'm testing against appear in my test.

Reply:
Doesn't look like `<code>` is required. I actually had an assert upfront to verify node type, but removed it before submitting. Putting it back in in https://github.com/StackExchange/Stacks-Editor/pull/72/commits/ffd77b340444d9a02c9b214c14f5b6fa8c25d968.